DSharpPlus/Entities/Channel/Message/DiscordMessage.cs

```diff
@@ -712,6 +721,57 @@ public async Task DeleteAllReactionsAsync(string? reason = null)
     public async Task DeleteReactionsEmojiAsync(DiscordEmoji emoji)
         => await this.Discord.ApiClient.DeleteReactionsEmojiAsync(this.ChannelId, this.Id, emoji.ToReactionString());
 
+    /// <summary>
+    /// Immediately ends the poll. You cannot end polls from other users.
+    /// </summary>
+    /// <returns></returns>
+    /// <exception cref="Exceptions.UnauthorizedException">Thrown when the client does not have the <see cref="DiscordPermissions.ManageMessages"/> permission or if the poll is not owned by the client.</exception>
+    /// <exception cref="Exceptions.NotFoundException">Thrown when the message does not have a poll.</exception>
+    /// <exception cref="Exceptions.BadRequestException">Thrown when an invalid parameter was provided.</exception>
+    /// <exception cref="Exceptions.ServerErrorException">Thrown when Discord is unable to process the request.</exception>
+    public async Task<DiscordMessage> EndPollAsync()
+        => await this.Discord.ApiClient.EndPollAsync(this.ChannelId, this.Id);
+
+
+    /// <summary>
+    /// Retrieves a full list of users that voted a specified answer on a poll. This will execute one API request per 100 entities.
+    /// </summary>
+    /// <param name="answerId">The id of the answer to get the voters of.</param>
+    /// <param name="cancellationToken">Cancels the enumeration before the next api request</param>
+    /// <returns>A collection of all users that voted the specified answer on the poll.</returns>
+    /// <exception cref="Exceptions.ServerErrorException">Thrown when Discord is unable to process the request.</exception>
+    public async IAsyncEnumerable<DiscordUser> GetAllPollAnswerVotersAsync
+    (
+        int answerId,
```

Comment:
Unsure if we should take in an answerId here or a DiscordPollAnswer object.

Reply:
I think id is more convenient because its a simple index, you could provide an overload if you want

Reply:
[It's not guaranteed to stay an index for the future](https://github.com/discord/discord-api-docs/pull/6746/files#diff-81a2c7d9f047834aa38e226fa4c7bf74b98831cd65116f76ea5281b21a9ce759R71-R73), but yea I'll ad an override.

Reply:
Yeah but even if they would make it an abitray id its easier to store it in a db for example, also i like the "we provide methods with the least required parameters" way